packages/yoastseo/spec/languageProcessing/researches/keywordCountSpec.js

```diff
@@ -25,221 +26,779 @@ const buildMorphologyMockResearcher = function( keyphraseForms ) {
 		expectedCount: 1,
-		expectedMarkings: [ new Mark( {
-			marked: "A string with a <yoastmark class='yoast-text-mark'>key</yoastmark>-<yoastmark class='yoast-text-mark'>word</yoastmark>.",
-			original: "A string with a key-word." } ) ],
-		// Note: this behavior might change in the future.
+		expectedMarkings: [ new Mark( { marked: "A string with a <yoastmark class='yoast-text-mark'>keyword</yoastmark>.",
+			original: "A string with a keyword.",
+			position: { endOffset: 26, startOffset: 19 } } ) ],
+		skip: false,
 	},
 	{
-		description: "counts a string with a keyword with a '_' in it",
-		paper: new Paper( "A string with a key_word.", { keyword: "key_word" } ),
-		keyphraseForms: [ [ "key_word", "key_words" ] ],
+		description: "with exact matching, a singular single word keyphrase should not be counted if the focus keyphrase is plural",
+		paper: new Paper( "<p>A string with a keyword.</p>", { keyword: "\"keywords\"" } ),
+		keyphraseForms: [ [ "keywords" ] ],
+		expectedCount: 0,
+		expectedMarkings: [],
+		skip: false,
+	},
+	{
+		description: "with exact matching, a multi word keyphrase should be counted if the focus keyphrase is the same",
+		paper: new Paper( "<p>A string with a key phrase key phrase.</p>", { keyword: "\"key phrase\"" } ),
+		keyphraseForms: [ [ "key phrase" ] ],
+		expectedCount: 2,
+		expectedMarkings: [ new Mark( { marked: "A string with a <yoastmark class='yoast-text-mark'>key phrase key phrase</yoastmark>.",
+			original: "A string with a key phrase key phrase.",
+			position: { endOffset: 40, startOffset: 19 } } ) ],
+		skip: false,
+	},
+	{
+		description: "with exact matching, a multi word keyphrase should be counted if the focus keyphrase is the same",
+		paper: new Paper( "<p>A string with a key phrase.</p>", { keyword: "\"key phrase\"" } ),
+		keyphraseForms: [ [ "key phrase" ] ],
 		expectedCount: 1,
-		expectedMarkings: [ new Mark( { marked: "A string with a <yoastmark class='yoast-text-mark'>key_word</yoastmark>.",
-			original: "A string with a key_word." } ) ],
+		expectedMarkings: [ new Mark( { marked: "A string with a <yoastmark class='yoast-text-mark'>key phrase</yoastmark>.",
+			original: "A string with a key phrase.",
+			position: { endOffset: 29, startOffset: 19 } } ) ],
+		skip: false,
 	},
 	{
-		description: "counts a string with with a 'ı' in the keyphrase",
-		paper: new Paper( "A string with with 'kapaklı' as a keyword in it", { keyword: "kapaklı" } ),
-		keyphraseForms: [ [ "kapaklı" ] ],
+		description: "with exact matching, a multi word keyphrase should not be counted if " +
+			"the focus keyphrase has the same words in a different order",
+		paper: new Paper( "<p>A string with a phrase key.</p>", { keyword: "\"key phrase\"" } ),
+		keyphraseForms: [ [ "key phrase" ] ],
+		expectedCount: 0,
+		expectedMarkings: [],
+		skip: false,
+	},
+	{
+		description: "with exact matching, it should match a full stop if it is part of the keyphrase and directly precedes the keyphrase.",
```

Comment:
TBH, in this issue I am not sure if we cover ALL cases with punctuation or special characters. Since we didn't even have those unit tests yet in the past and we also don't know what is the expected behavior. Marina created a separate tech debt issue to cover this: https://github.com/Yoast/lingo-other-tasks/issues/183

The cases that are already added in this PR are for the cases we already know what is the expected behavior.

Reply:
But yes I can still add this one other case you mentioned above :D